packages/desktop-client/src/components/reports/graphs/LineGraph.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+
+import { css } from 'glamor';
+import {
+  LineChart,
+  Line,
+  CartesianGrid,
+  XAxis,
+  YAxis,
+  Tooltip,
+  ResponsiveContainer,
+} from 'recharts';
+
+import { theme } from '../../../style';
+import { type CSSProperties } from '../../../style';
+import AlignedText from '../../common/AlignedText';
+import PrivacyFilter from '../../PrivacyFilter';
+import Container from '../Container';
+
+type PayloadItem = {
+  payload: {
+    date: string;
+    assets: number | string;
+    debt: number | string;
+    networth: number | string;
+    change: number | string;
+  };
+};
+
+type CustomTooltipProps = {
+  active?: boolean;
+  payload?: PayloadItem[];
+  label?: string;
+};
+
+const CustomTooltip = ({ active, payload, label }: CustomTooltipProps) => {
+  if (active && payload && payload.length) {
+    return (
+      <div
+        className={`${css({
+          zIndex: 1000,
+          pointerEvents: 'none',
+          borderRadius: 2,
+          boxShadow: '0 1px 6px rgba(0, 0, 0, .20)',
+          backgroundColor: theme.menuAutoCompleteBackground,
+          color: theme.menuAutoCompleteText,
+          padding: 10,
+        })}`}
+      >
+        <div>
+          <div style={{ marginBottom: 10 }}>
+            <strong>{payload[0].payload.date}</strong>
+          </div>
+          <div style={{ lineHeight: 1.5 }}>
+            <PrivacyFilter>
+              <AlignedText left="Assets:" right={payload[0].payload.assets} />
+              <AlignedText left="Debt:" right={payload[0].payload.debt} />
+              <AlignedText
+                left="Change:"
+                right={<strong>{payload[0].payload.change}</strong>}
+              />
+            </PrivacyFilter>
+          </div>
+        </div>
+      </div>
+    );
+  }
+};
+
+type LineGraphProps = {
+  style?: CSSProperties;
+  graphData;
+  compact: boolean;
+  domain?: {
+    y?: [number, number];
+  };
+};
+type PotentialNumber = number | string | undefined | null;
+
+const numberFormatterTooltip = (value: PotentialNumber): number | null => {
+  if (typeof value === 'number') {
+    return Math.round(value);
+  }
+  return null; // or some default value for other cases
+};
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I'm seeing this util duplicated in a bunch of components. Maybe worth extracting to a reusable util?